microsoft/testsuites/vm_extensions/run_command.py

```diff
@@ -0,0 +1,295 @@
+        2. Provided a custom linux shell script
+        3. Provided a public storage blob uri that points to the script
+        4. Provided a storage uri pointing to script without a sas token (should fail)
+        5. Provided a storage sas uri that points to script
+    """,
+)
+class RunCommand(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a pre-existing ifconfig script.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_existing_script_run(self, log: Logger, node: Node) -> None:
+        settings = {"source": {"CommandId": "ifconfig"}}
+
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="RunCommand",
+            publisher="Microsoft.CPlat.Core",
+            type_="RunCommandHandlerLinux",
+            type_handler_version="1.3",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+        )
+
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a custom shell script.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_custom_script_run(self, log: Logger, node: Node) -> None:
+        test_file = f"/tmp/{str(uuid.uuid4())}"
+        settings = {
+            "source": {"CommandId": "RunShellScript", "script": f"touch {test_file}"}
+        }
+
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="RunCommand",
+            publisher="Microsoft.CPlat.Core",
+            type_="RunCommandHandlerLinux",
+            type_handler_version="1.3",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+        )
+
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        message = f"File {test_file} was not created on the test machine"
+        # Verify that file was created on the test machine
+        node.execute(
+            f"ls '{test_file}'",
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=message,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a public uri pointing to the
+        script in blob storage.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_public_uri_script_run(self, log: Logger, node: Node) -> None:
+        script_uri = "https://rcv2lisa.blob.core.windows.net/publiccontainer/test.sh"
```

Comment:
It's not recommended to use a public blob uri. I don't know if it can pass the security requirements. And I'm not sure if it extends the coverage.